extension/js/common/platform/store/contact-store.ts

```diff
@@ -380,12 +380,940 @@ export class ContactStore extends AbstractStore {
     });
   };
 
+
+  // taken from https://github.com/datasets/top-level-domain-names/blob/master/top-level-domain-names.csv
+  private static tlds = [
```

Comment:
if we don't want to index top-level domains, we don't have to list them.
as a e-mail (example: `com@example.com`) now breaks into these index entries:
1) `com@example.com`
2) `example.com`
3) `com`
You see, the top-level domain is the last entry of the resulting array, so it is simple to exclude it:
`Str.splitAlphanumericExtended(email).slice(0, -1)`
As for _common domains_. we can create and use a list like this the way you did.

Reply:
@rrrooommmaaa Fixed it.

Reply:
Anything else should be done here? Where to put this list of common domains and how to use it for filtering?

Reply:
The way you did previously:
const commonDomains = [
'gmail.com'
...
]

```
.filter(s => !ContactStore.commonDomains.includes(ContactStore.normalizeString(s)));
```

Also, need to update this procedure
https://github.com/FlowCrypt/flowcrypt-browser/blob/42d87e826decfe663940972ad60a518bacbf00df/extension/js/background_page/migrations.ts#L143
It currently updates all searchables (and sets the flag `contact_store_searchable_extended` in order to not be run twice).
As we now need to run the update even if this flag is set, we need to add a new flag like `contact_store_searchable_pruned`. The old flag `contact_store_searchable_extended` should be removed because we no longer need it (if `searchable` wasn't extended, it will be during this migration -- `updateSearchable()` fully re-generates the `searchable`).
And the method itself should be renamed to `updateSearchables`.